src/Libraries/LibAllowList.sol

```diff
@@ -71,13 +69,51 @@ library LibAllowList {
     /// @dev Add a selector to the allow list
     /// @param _selector the selector to add
     function addAllowedSelector(bytes4 _selector) internal {
-        _getStorage().selectorAllowList[_selector] = true;
+        AllowListStorage storage als = _getStorage();
+
+        if (als.selectorAllowList[_selector]) {
+            // Check if selector exists in the array to handle legacy state
+            // where a selector might be true in selectorAllowList mapping but missing from the array.
+            // Since this is an admin-only function called infrequently, the array iteration is acceptable.
```

Comment:
If we re-sync all our selectors and start fresh, why do we need legacy handling? Wondering if this is an unnecessary gas consumption (that happens on every sync).

Reply:
I would say we could probably remove all variables associated with the old layout if not needed in the new layout and update the namespace so the old whitelist is no longer referenced. Thoughts?

Reply:
I like this approach the most @ezynda3

Reply:
Btw are all sigs.json the same for all chains?
>> at the moment yes but this will change in the near future

and fine with Ed's approach